### PR TITLE
added hop volume aggregator

### PR DESCRIPTION
**Pull request by bonkman22**
**Name (to be shown on DefiLlama):**
Hop Aggregator

**Twitter Link:**
https://x.com/hopaggregator

**Website Link:**
https://hop.ag/

**Logo (High resolution, will be shown with rounded borders):**
https://hop.ag/hop_logos/inverse_logo.png

**Chain:**
SUI

**Short Description (to be shown on DefiLlama):**
Hop Aggregator is a highly optimized DEX aggregator for the Sui ecosystem with 0 fees. By quickly identifying the optimal routes for trades of any size, hop simplifies the Sui DeFi experience.

**Category (full list at https://defillama.com/categories) *Please choose only one:**
DEX Aggregator

**Github org/user (Optional, if your code is open source, we can track activity):**
https://github.com/hopaggregator

### CODE REVIEW

**Comment by llamatester** [comment]
The hop-aggregator adapter exports: 
        
 
 _______________________________________
    Aggregators for 22/5/2024
    _______________________________________
    
    SUI 👇
    Backfill start time: 22/5/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 957.22 k
    Total volume: 70.19 M
    Daily fees: 0
    Total fees: 0
    Timestamp: 1716407567 (2024-05-22T19:52:47.000Z)

**Comment by llamatester** [comment]
The hop-aggregator adapter exports: 
        
 
 _______________________________________
    Aggregators for 24/5/2024
    _______________________________________
    
    SUI 👇
    Backfill start time: 24/5/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 2.91 M
    Total volume: 79.78 M
    Daily fees: 0
    Total fees: 0
    Timestamp: 1716538474 (2024-05-24T08:14:34.000Z)

**Comment by dtmkeng** [comment]
can't list adapter return value 0 seem you fix dailyVolume is 0 could you check if can get daily data

File: aggregators/hop-aggregator/index.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+const fetchVolume = async (options: FetchOptions) => {
+  let timestamp = options.startOfDay * 1_000;
+
+  const res = await httpPost("http://35.153.229.202/api/volume", { timestamp });
+  const record = res.record || {};
+
+  return {
+    dailyVolume: record.total_volume_usd ? record.total_volume_usd.toFixed(2) : 0,
+    totalVolume: res.total_volume_usd ? res.total_volume_usd.toFixed(2) : 0,
+    dailyFees: 0,
```
Comment (dtmkeng) [inline aggregators/hop-aggregator/index.ts:14]: could you get daily value for fees and volume ?

**Review by dtmkeng — CHANGES_REQUESTED** [review]

File: aggregators/hop-aggregator/index.ts (hunk 1 above)
Comment (bonkman22) [reply aggregators/hop-aggregator/index.ts:14]: Yes, this is already working!

We charge 0 fees and will not change this.

As for the dailyVolume, in the response from the server we return it in the format ```{ total_volume_usd: 100, record: { total_voume_usd: 100 }}```. The record itself is for the day. Meanwhile, the first total_volume_usd is for overall.

**Pull request merged**